core/serialization/blocks.js

```diff
@@ -263,8 +263,162 @@ const saveConnection = function(connection) {
  * @param {!State} state The state of a block to deserialize into the workspace.
  * @param {!Workspace} workspace The workspace to add the block to.
  */
-// eslint-disable-next-line no-unused-vars
 const load = function(state, workspace) {
-  // Temporarily NOP while connecting things together.
+  loadInternal(state, workspace);
+};
+
+/**
+ * Loads the block represented by the given state into the given workspace.
+ * This is defined internally so that the extra optional parameter doesn't
+ * clutter our external API.
+ * @param {!State} state The state of a block to deserialize into the workspace.
+ * @param {!Workspace} workspace The workspace to add the block to.
+ * @param {!Connection} parentConnection The optional parent connection to
+ *     attach the block to.
+ */
+const loadInternal = function(state, workspace, parentConnection = undefined) {
+  const block = workspace.newBlock(state['type'], state['id']);
+  loadCoords(block, state);
+  loadAttributes(block, state);
+  loadExtraState(block, state);
+  if (parentConnection &&
+      (block.outputConnection || block.previousConnection)) {
+    parentConnection.connect(
+        /** @type {!Connection} */
+        (block.outputConnection || block.previousConnection));
+  }
+  // loadIcons(block, state);
```

Comment:
Commented out code should be deleted.

Reply:
This gets uncommented in #5216 when I add serialization of comments, so I'm just going to leave it for now.